hail/python/hailtop/aiotools/tasks.py

```diff
@@ -21,9 +21,19 @@ def ensure_future(self, coroutine):
         if self._closed:
             raise TaskManagerClosedError
         t = asyncio.create_task(coroutine)
-        t.add_done_callback(lambda _: self.tasks.remove(t))
+        t.add_done_callback(self.on_task_done(t))
         self.tasks.add(t)
 
+    def on_task_done(self, t: asyncio.Task) -> Callable[[asyncio.Future], None]:
+        def callback(fut: asyncio.Future):
+            self.tasks.remove(t)
+            try:
+                if e := fut.exception():
+                    log.exception(e)
+            except asyncio.CancelledError:
```

Comment:
log the cancellation too?

Reply:
It's ok to be cancelled on worker shutdown, so I've only logged if that's not the case.